modules/javafx.controls/src/main/java/javafx/scene/control/skin/ProgressIndicatorSkin.java

```diff
@@ -129,9 +131,11 @@ public ProgressIndicatorSkin(ProgressIndicator control) {
         // register listeners
         registerChangeListener(control.indeterminateProperty(), e -> initialize());
         registerChangeListener(control.progressProperty(), e -> updateProgress());
-        registerChangeListener(NodeHelper.treeShowingProperty(control), e -> updateAnimation());
         registerChangeListener(control.sceneProperty(), e->updateAnimation());
 
+        treeShowingExpression = new TreeShowingExpression(control);
+        treeShowingExpression.addListener((obs, old, current) -> updateAnimation());
```

Comment:
Is there a reason not to still use `registerChangeListener` for this?

Reply:
`registerChangeListener` can be used here, I just didn't see it.  However, there is no need to unregister the listener from the expression (it is local to the skin, it will never have more than one listener and will be GC'd with the Skin).  I will change it though, as it looks a bit cleaner, and there is no harm unregistering the listener.